### PR TITLE
Working with CSV files in Java

### LINKED ISSUE #5043: Working with CSV files in java
#### Topic Suggestion
Working with CSV files in java 
#### Proposed title of article
Working with CSV files in java
#### Proposed article introduction
The CSV represents Comma-Separated Values. It is a basic document design which is utilized to store plain information in straightforward text structure, like an accounting page or data set. The documents in the CSV organization can be imported to and sent out from programs (Microsoft Office and Excel) which store information in tables. The CSV document utilized a delimiter to recognize and isolate various information token in a record. The CSV record design is utilized when we move plain information between programs that locally work on incongruent organizations.

#### Key takeaways 
- Step by step instructions to make CSV document, utilizing notepand and Microsoft Excel.
- Java String.split() technique. 
- Java scanner class. 
- Using open CSV API.

#### Article quality
In this article I will acquaint client with CSV documents utilizing java,am additionally going to take the client through the strategies of making a csv record utilizing note pad and Microsoft Excel, likewise I will disclose to the client how CSV records are perused and deciphered utilizing various techniques. I trust the client will appreciate to know how csv documents work in java.
#### References
N/A

### PR DESCRIPTION
**Pull request by tesswambui**
#### Formatting and structure checks
- [X] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [X] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [X] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [X] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [X] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [X] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [X] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [X] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [X] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [X] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [X] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [X] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [X] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [X] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [X] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [X] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

#### Topic suggestion this closes
This closes #5043

### CODE REVIEW

**Comment by Kayere** [comment]
Hello @tesswambui 

 Upon running your article through our 3rd party plagiarism checker it seemed to raise a few flags and the % was higher than we typically accept.

Please see attached PDF - and revisit the article to ensure we are contributing wholly unique and original content.

Be sure to see our resources page to see more info on plagiarism and what is considered as such.

[Working with CSV files in Java #5481 Copyleaks Report.pdf](https://github.com/section-engineering-education/engineering-education/files/7693179/Working.with.CSV.files.in.Java.5481.Copyleaks.Report.pdf)

**Comment by paulodhiambo** [comment]
@tesswambui please delete the index.md file here `content/articles/index.md` you have two article index.md files.

**Comment by tesswambui** [comment]
Hello @paulodhiambo. I have made the change you requested, kindly check. Thanks

File: content/articles/index.md [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+
+OpenCSV is used since Java does not provide native support to read Comma-Separated values.
+
+#### Important OpenCSV classes
+- `CSVWrite` -  enables writing data into a CSV file.
+- `CSVToBean` - utilized to populate Java Application with CSV data.
+- `BeanToCSV` - applied to export information from JavaBean to Comma-seperated value file.
+Comma-separated values are read in two ways, namely:
+- By line: In this case, the comma-separated values are consecutively read from one line to another.
+- All data: utilizes readAll() technique to peruse all records at once.
```
Comment (paulodhiambo) [inline content/articles/index.md:141]: Are these two also openCSV classes?

**Review by paulodhiambo — CHANGES_REQUESTED** [review]

File: content/articles/index.md (hunk 1 above)
Comment (tesswambui) [reply content/articles/index.md:141]: @paulodhiambo . The two  are  openCSV classes. Those  ways of reading a CSV file in java.

**Comment by hectorkambow** [comment]
jumping on this one

**Comment by hectorkambow** [comment]
@tesswambui  

I believe there is an image missing from the files. 

cc @paulodhiambo

**Comment by hectorkambow** [comment]
semi-colon-seperated.png

**Comment by tesswambui** [comment]
Hello @hectorkambow, @paulodhiambo  , I uploaded the missing image thank.

**Comment by paulodhiambo** [comment]
Hello @tesswambui please upload the image to the correct directory

**Comment by tesswambui** [comment]
Hae @paulodhiamb, I have made the changes by uploading the image to the correct directory.

**Comment by paulodhiambo** [comment]
Hello @tesswambui please delete "content/articles/semi-colon-seperated.png"

**Comment by lalith1403** [comment]
jumping on this one 🚀

**Pull request merged**